src/components/PostDetails/PostDetails.jsx

```diff
@@ -1,45 +1,112 @@
-import React from 'react';
-import { NewCommentForm } from '../NewCommentForm';
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
 import './PostDetails.scss';
 
-export const PostDetails = () => (
-  <div className="PostDetails">
-    <h2>Post details:</h2>
-
-    <section className="PostDetails__post">
-      <p>sunt aut facere repellat provident occaecati excepturi optio</p>
-    </section>
-
-    <section className="PostDetails__comments">
-      <button type="button" className="button">Hide 2 comments</button>
-
-      <ul className="PostDetails__list">
-        <li className="PostDetails__list-item">
-          <button
-            type="button"
-            className="PostDetails__remove-button button"
-          >
-            X
-          </button>
-          <p>My first comment</p>
-        </li>
-
-        <li className="PostDetails__list-item">
-          <button
-            type="button"
-            className="PostDetails__remove-button button"
-          >
-            X
-          </button>
-          <p>sad sds dfsadf asdf asdf</p>
-        </li>
-      </ul>
-    </section>
-
-    <section>
-      <div className="PostDetails__form-wrapper">
-        <NewCommentForm />
-      </div>
-    </section>
-  </div>
-);
+import { NewCommentForm } from '../NewCommentForm';
+import { getPostDetails } from '../../api/posts';
+import { addComment, removeComment, getPostComments } from '../../api/comments';
+
+export const PostDetails = ({ selectedPostId }) => {
+  const [userPost, setUserPost] = useState(0);
+  const [postComments, setPostComments] = useState([]);
+  const [showComments, setShowComments] = useState(true);
+
+  useEffect(() => {
+    getPostDetails(selectedPostId).then(setUserPost);
+
+    fetchPostComments();
+  }, [selectedPostId]);
+
+  const fetchPostComments = () => {
+    getPostComments(selectedPostId)
+      .then(commentsFromServer => setPostComments(
+        commentsFromServer,
+      ));
+
+    getPostComments(selectedPostId);
+  };
+
+  const deleteSelectedComment = (commentId) => {
+    removeComment(commentId);
+
+    setPostComments(comments => comments
+      .filter(comment => comment.id !== commentId));
```

Comment:
Let me explain what I meant.
You have your 'postComments' in your state. Initially, it's an empty array (because there are no comments loaded).
On componentDidMount you are fetching your comments from the server ('fetchPostComments' function) and then they are displayed in your app.
After you DELETE some comment (in the 'deleteSelectedComment' method) you ACTUALLY delete it from the server (you sending a request with DELETE method). So you should re-fetch the comments again (because they are changed) to get the actual list of comments from the server. So after deleting you should immediately fetch new comments again.
Now you're just filtering locally already fetched before comments and in your list, they are not synchronized with the server.

Reply:
I understood you yesterday, just my mind left me at midnight, thanks!